gradoop-accumulo/src/main/java/org/gradoop/common/storage/impl/accumulo/iterator/GradoopEdgeIterator.java

```diff
@@ -0,0 +1,118 @@
+ * limitations under the License.
+ */
+
+package org.gradoop.common.storage.impl.accumulo.iterator;
+
+import javafx.util.Pair;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.gradoop.common.storage.impl.accumulo.constants.AccumuloTables;
+import org.gradoop.common.storage.impl.accumulo.row.EdgeRow;
+import org.gradoop.common.utils.Json;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * accumulo gradoop edges iterator
+ */
+public class GradoopEdgeIterator extends BaseElementIterator<EdgeRow> {
+
+  @Nonnull
+  @Override
+  public EdgeRow fromRow(@Nonnull Map.Entry<Key, Value> pair) {
+    //map from serialize content
+    EdgeRow content = Json.loads(pair.getValue().get(), EdgeRow.class);
+    content.setId(pair.getKey().getRow().toString());
+    //read from content
+    return content;
+  }
+
+  @Nonnull
+  @Override
+  public Pair<Key, Value> toRow(@Nonnull EdgeRow record) {
+    //write to content
+    return new Pair<>(new Key(record.getId()), new Value(Json.dumps(record)));
+  }
+
+  @Nonnull
+  @Override
+  public List<EdgeRow> doSeek(SortedKeyValueIterator<Key, Value> source, Range range) throws
+    IOException {
+    List<EdgeRow> result = new ArrayList<>();
+    do {
+      EdgeRow edge = readLine(source);
+      if (edge != null) {
+        result.add(edge);
+      }
+    } while (source.hasTop());
+    return result;
+  }
+
+  /**
+   * read next edge element from table store
+   * @param source origin accumulo source
+   * @return edge element
+   * @throws IOException io err
+   */
+  @Nullable
+  private EdgeRow readLine(SortedKeyValueIterator<Key, Value> source) throws IOException {
+    EdgeRow row = new EdgeRow();
+    if (!source.hasTop()) {
+      return null;
+    }
+    row.setId(source.getTopKey().getRow().toString());
+    while (source.hasTop()) {
+      Key key = source.getTopKey();
+      Value value = source.getTopValue();
+
+      if (!Objects.equals(row.getId(), key.getRow().toString())) {
+        break;
+      }
+
+      switch (key.getColumnFamily().toString()) {
+      case AccumuloTables.KEY.LABEL:
+        row.setLabel(value.toString());
+        break;
+      case AccumuloTables.KEY.SOURCE:
+        row.setSource(value.toString());
+        break;
+      case AccumuloTables.KEY.TARGET:
+        row.setTarget(value.toString());
+        break;
+      case AccumuloTables.KEY.PROPERTY:
+        row.getProperties().put(key.getColumnQualifier().toString(), value.toString());
+        break;
+      case AccumuloTables.KEY.GRAPH:
+        row.getGraphs().add(key.getColumnQualifier().toString());
+        break;
+      default:
+        break;
```

Comment:
the default case should never happen, right?

Reply:
I can't change this, this is required by style-check plugin

```
[INFO] --- maven-checkstyle-plugin:3.0.0:check (default) @ gradoop-accumulo ---
[INFO] Starting audit...
[ERROR] /Users/chpengzh/Documents/opensource/gradoop/gradoop-accumulo/src/main/java/org/gradoop/common/storage/impl/accumulo/iterator/tserver/GradoopGraphHeadIterator.java:93: switch without "default" clause. [MissingSwitchDefault]
[ERROR] /Users/chpengzh/Documents/opensource/gradoop/gradoop-accumulo/src/main/java/org/gradoop/common/storage/impl/accumulo/iterator/tserver/GradoopVertexIterator.java:93: switch without "default" clause. [MissingSwitchDefault]
[ERROR] /Users/chpengzh/Documents/opensource/gradoop/gradoop-accumulo/src/main/java/org/gradoop/common/storage/impl/accumulo/iterator/tserver/GradoopEdgeIterator.java:93: switch without "default" clause. [MissingSwitchDefault]
Audit done.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```